### PR TITLE
Deduplicate x86_64 test execution and rename coverage job to test

### PR DESCRIPTION
**Pull request by sandersaares**
## Problem

On x86_64 (ubuntu / macos / windows) the test suite was executed **twice** per platform:

- `test-more-x64` (MSRV): `nextest run --tests` + `cargo test --benches`
- `coverage` (nightly): `llvm-cov nextest --lib --bins --tests --examples` — a *superset* of the `--tests` part above, just with coverage instrumentation.

So every unit/integration test ran once for results and once for coverage. As you noted, gathering coverage is merely a side effect of running the tests, which also made the `coverage` job name misleading.

## Fix

Merge the two x86_64 jobs into a single `test-x64` job and rename the ARM counterpart for symmetry.

- **`coverage` → `test-x64`**: runs the full instrumented suite (the existing `coverage-measure` + `coverage-report-ci`), then runs the benchmark panic-check once. Uploads both the coverage report (lcov) and the test results (junit) to Codecov. It stays on the **nightly** toolchain because `llvm-tools-preview` is installed only for nightly (`just install-tools`), so coverage cannot move to MSRV without a larger toolchain change.
- **Removed `test-more-x64`**: its `--tests` run was the duplicate (already covered by the instrumented run); its `cargo test --benches` smoke-check is folded into `test-x64` via a new `test-benches` recipe.
- **`test-more-arm` → `test-arm`**: unchanged behaviour. It remains the **MSRV** test pass and the ARM-gated-code pass. No coverage here — instrumentation is x86_64-only, as before.
- Repointed dependents: `miri-x64`, `mutants`, `careful` → `test-x64`; `miri-arm` → `test-arm`.
- Updated the in-workflow comments and `.github/workflows/AGENTS.md` to match.

## Net effect on toolchain coverage

No platform runs the same tests twice, while both toolchains are still exercised:

- x86_64 unit / integration / example tests: **nightly** (`test-x64`, with coverage)
- x86_64 benchmark panic-check: **MSRV** (`test-x64` → `test-benches`)
- ARM tests + benchmark panic-check: **MSRV** (`test-arm`)
- MSRV compilation of every target: `check-dev`

## Verification

- Parsed `validation.yml` and confirmed every `needs:` resolves to a defined job (no dangling references to the old job names).
- `just --list` parses; the new `test-benches` recipe is registered.
- Ran `coverage-measure` locally and confirmed `llvm-cov nextest` still writes junit to `target/nextest/default/junit.xml`, so the test-results upload path is unchanged.
- Ran `test-benches` locally to confirm the recipe executes.

### CODE REVIEW

File: justfiles/just_testing.just [hunk 1]
```diff
@@ -347,20 +347,30 @@ test-azure ACCOUNT=env_var_or_default('BENCH_HISTORY_TEST_AZURE_ACCOUNT', ''):
...
+test-benches FILTER="":
+    $ErrorActionPreference = "Stop"
+    $PSNativeCommandUseErrorActionPreference = $true
+
+    cargo +$env:RUST_MSRV test {{ target_package }} --all-features {{ FILTER }} --locked --no-fail-fast --benches
+
+# Runs regular tests as well as doing one iteration of benchmarks (the latter via
+# `test-benches`). We separate it from "test" because benchmarks are quite slow, so we tend
+# to only run them as part of a full validation pass. Tests go through nextest (parallel
+# execution, junit reporting, retries, etc.).
```
[the remark this answers is not in the document]
Comment (sandersaares) [reply justfiles/just_testing.just]: Fixed in 73c19edd. 	est-more now runs the nextest tests in its body and delegates the bench smoke-check to 	est-benches as a && post-dependency, so the command lives in one place and the comment is accurate. Execution order is preserved (nextest tests first, then benches).

File: .github/workflows/validation.yml [hunk 2]
```diff
@@ -165,7 +165,17 @@ jobs:
...
+  # The x86_64 test pass. Runs the full test suite (unit, integration and example tests)
+  # with coverage instrumentation, so the coverage report is a side effect of the regular
+  # test run rather than a second, separately re-executed pass. Also runs the benchmark
+  # targets once to verify they do not panic. Uploads both the coverage report and the
+  # test results to Codecov.
+  #
+  # Coverage requires the nightly toolchain (llvm-tools-preview is installed only there via
+  # `just install-tools`), so this is the nightly test pass. The MSRV test pass runs on the
+  # ARM runners (`test-arm`), and MSRV compilation of every target is covered by `check-dev`.
+  # Paired with `test-arm`; both carry an explicit `-x64`/`-arm` suffix for symmetry.
```
[the remark this answers is not in the document]
Comment (sandersaares) [reply .github/workflows/validation.yml]: Good catch. GitHub's macos-latest is now Apple Silicon (macos-26/arm64), so testing macOS under an `x64` label was miscategorizing it. Per maintainer guidance, macOS moved into the arm64 jobs: 	est-x64 is now ubuntu-latest + windows-latest (genuinely x86_64) and 	est-arm is now ubuntu-24.04-arm + windows-11-arm + macos-latest (genuinely arm64). Fixed in 0a10300b; job names and docs updated to match.

File: .github/workflows/AGENTS.md [hunk 3]
```diff
@@ -26,15 +26,27 @@ Split from the monolithic `just validate-local` into individual jobs:
 - **Multi-platform jobs** (run on ubuntu-latest, macos-latest, windows-latest):
   - check-dev
   - clippy-dev
-  - test-more-x64
+  - **test-x64** — the x86_64 test pass (ubuntu-latest, macos-latest, windows-latest)
```
[the remark this answers is not in the document]
Comment (sandersaares) [reply .github/workflows/AGENTS.md]: Good catch. GitHub's macos-latest is now Apple Silicon (macos-26/arm64), so testing macOS under an `x64` label was miscategorizing it. Per maintainer guidance, macOS moved into the arm64 jobs: 	est-x64 is now ubuntu-latest + windows-latest (genuinely x86_64) and 	est-arm is now ubuntu-24.04-arm + windows-11-arm + macos-latest (genuinely arm64). Fixed in 0a10300b; job names and docs updated to match.
